app/azure_logging.py

```diff
@@ -45,22 +72,20 @@ def initialize_logging(environment: str, logging_level: int, correlation_id: Opt
         azurelog_handler.addFilter(ExceptionTracebackFilter())
         logger.addHandler(azurelog_handler)
     except ValueError as e:
-        if environment == "local":
-            logger.info(f"Application Insights logger handler not set.: {e}") 
+        if os.environ.get("APPLICATIONINSIGHTS_CONNECTION_STRING") is None:
```

Comment:
We discussed this in a previous PR.
The problem with this is that it will fail silently, in production, if the environment variable is unset.
The reason we decided to special case local development was because, in all other situations, we want this to fail (at least, that was where the discussion started)

Reply:
Yeah, I was torn on this. The trouble is (as always!) naming – this PR adds `ENVIRONMENT=dev` that gets served from your local machine. But then you can't have `environment in ("local", "dev")` because a `dev` may get deployed in CI. So either need to add a flag that is like `SERViNG_FROM_LOCAL=True` or something, which is a bit horrible, or accept that `APPLICATIONINSIGHTS_CONNECTION_STRING` needs to be set in prod. I don't see a scenario where it wouldn't be: https://github.com/UCLH-Foundry/FlowEHR/blob/1d21ea13ff8a974a55887fd7abdfa47d4cb9b065/apps/app/webapp.tf#L49